src/sql/src/normalize.rs

```diff
@@ -388,39 +388,44 @@ pub fn create_statement(
 
 macro_rules! with_option_type {
     ($name:ident, String) => {
-        if let Some(crate::ast::WithOptionValue::Value(crate::ast::Value::String(value))) = $name {
-            value
-        } else if let Some(crate::ast::WithOptionValue::ObjectName(name)) = $name {
-            crate::ast::display::AstDisplay::to_ast_string(&name)
-        } else {
-            ::anyhow::bail!("expected String");
+        match $name {
+            Some(crate::ast::WithOptionValue::Value(crate::ast::Value::String(value))) => value,
+            Some(crate::ast::WithOptionValue::ObjectName(name)) => {
+                crate::ast::display::AstDisplay::to_ast_string(&name)
+            }
+            _ => ::anyhow::bail!("expected String"),
         }
     };
     ($name:ident, bool) => {
-        if let Some(crate::ast::WithOptionValue::Value(crate::ast::Value::Boolean(value))) = $name {
-            value
-        } else if $name.is_none() {
+        match $name {
+            Some(crate::ast::WithOptionValue::Value(crate::ast::Value::Boolean(value))) => value,
             // Bools, if they have no '= value', are true.
-            true
-        } else {
-            ::anyhow::bail!("expected bool");
+            None => true,
+            _ => ::anyhow::bail!("expected bool"),
         }
     };
     ($name:ident, Interval) => {
-        if let Some(crate::ast::WithOptionValue::Value(Value::String(value))) = $name {
-            ::repr::strconv::parse_interval(&value)?
-        } else if let Some(crate::ast::WithOptionValue::Value(Value::Interval(interval))) = $name {
-            ::repr::strconv::parse_interval(&interval.value)?
-        } else {
-            ::anyhow::bail!("expected Interval");
+        match $name {
+            Some(crate::ast::WithOptionValue::Value(Value::String(value))) => {
+                ::repr::strconv::parse_interval(&value)?
+            }
+            Some(crate::ast::WithOptionValue::Value(Value::Interval(interval))) => {
+                ::repr::strconv::parse_interval(&interval.value)?
+            }
+            _ => ::anyhow::bail!("expected Interval"),
```

Comment:
Why did these hunks all change? Just for aesthetics? In the future, please try to keep this stuff to a separate commit, or call it out in the code review as pure code movement. It's hard to review PRs that contain functionality changes, random doc comment improvements, and minor, functionality-preserving aesthetic changes all in the same commit. On my first pass I thought you changed the behavior of this macro, added documentation for the new behavior, and then used that new behavior, but it seems that's not the case.

Reply:
Ah, I intended to do that, my commits after this comment were made before I saw your response here, fwiw. It got moved to #5869 and added you explicitly as a reviewer on it.